packages/dict/src/dict.ts

```diff
@@ -0,0 +1,54 @@
+import { Predicate, Guard, Reducer, UnaryFn } from '@proem/function'
+
+export interface Dictionary {
+  [key:string]: any
```

Comment:
Is this a reasonable type for a dictionary? Using `any` loses a lot of type safety, and in my opinion, you very rarely want to put different types in a dictionary so fixing the type to a generic `T` would make a lot of sense.